tracee/tracee.go

```diff
@@ -71,6 +72,21 @@ type BoolFilter struct {
 	Enabled bool
 }
 
+type ArgFilter struct {
+	Filters map[ArgFilterKey]ArgFilterVal
```

Comment:
looks like you've chosen a design that's similar to the ebpf but since we are evaluating this in go it could be simper, just compare strings. This will allow us to implement filters like startswith which is not really important for filtering on context fields, but for filtering on args it's very practical (e.g pathname starts with /etc)

Reply:
I did want to check by prefix by decided to postpone it for a later commit. We also need it for vfs_write events filtering to get rid of the --filter-file-write - but that's another story.
I chose this implementation method because it is the most efficient one. Naive string comparison will require to loop over the available argfilters for every event received, which is not efficient. Let's keep it efficient for now and think about how to tackle the prefix issue in a different commit

Reply:
actually, since we operate on the `RawEvent`, you have the `RawArgs` map so don't need to iterate, right?

Reply:
Not really. Say that you gave a filter to filter all the pathname given to execve. Then you will have to compare "pathname" (deoded from the actual argument) against all the argument filters given. Here a map is used to avoid this loop

Reply:
I'm not looking at the code, but iirc you can encode the value into argtag, then look it up in the RawArgs map, no iteration involved, right?

Reply:
So you suggest iterating over all given arg filters instead of iterating the event's arguments?

Reply:
yes. it will probably be less iterations on average, but that's not my point. I suggest this only if you think it will make the code simpler considering this in go and considering your future plans for arg filtering

Reply:
But doesn't this loose the purpose you started with? To compare by string or prefix of strings?
If we need to encode the value into argtag, a prefix of the string will not work

Reply:
Not sure I understand what do you get by changing what you loop on.
I DO compare strings! Have a look at lines 845 and 855

Reply:
yes you are right it's actually just the order or iterations (still makes more sense to iterate on filters first, but not a big deal)

Reply:
Yes, it will also simplify things - I'm on it